File: SOC/src/cc/game/soc/core/EventCard.java
package cc.game.soc.core;

public class EventCard extends Card {

    static {
        addAllFields(EventCard.class);
    }

	private int production=-1;

    public EventCard() {}

	public EventCard(EventCardType t, int production) {
		super(t);
		this.production = production;
	}

	public final int getProduction() {
		return production;
	}

	public EventCardType getType() {
		return EventCardType.values()[getTypeOrdinal()];
	}
	
}

File: SOC/src/cc/game/soc/core/EventCard.kt
package cc.game.soc.core

class EventCard(type: EventCardType=EventCardType.NoEvent, val production: Int=-1) : Card(CardType.Event, type.ordinal, CardStatus.UNUSABLE) {
	companion object {
		init {
			addAllFields(EventCard::class.java)
		}
	}

	val type: EventCardType
		get() = EventCardType.values()[typeOrdinal]
}